modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/blueprint/search/request/enhancer/SXPBlueprintSearchRequestEnhancerImpl.java

```diff
@@ -395,6 +401,45 @@ private void _processElementInstances(
 		}
 	}
 
+	private ElementInstance[] _processIndexConfiguration(
+		Configuration configuration, ElementInstance[] elementInstances) {
+
+		if (configuration == null) {
+			return elementInstances;
+		}
+
+		IndexConfiguration indexConfiguration =
+			configuration.getIndexConfiguration();
+
+		if (ArrayUtil.isEmpty(elementInstances) ||
+			(indexConfiguration == null) ||
+			Validator.isBlank(indexConfiguration.getIndexName())) {
+
+			return elementInstances;
+		}
+
+		List<ElementInstance> elementInstanceList = new ArrayList<>(
+			elementInstances.length);
+
+		for (ElementInstance elementInstance : elementInstances) {
+			SXPElement sxpElement = elementInstance.getSxpElement();
+
+			if ((sxpElement != null) && sxpElement.getReadOnly()) {
+				ElementDefinition elementDefinition =
+					sxpElement.getElementDefinition();
+
+				if (Objects.equals("custom", elementDefinition.getCategory())) {
+					elementInstanceList.add(elementInstance);
+				}
+			}
+			else {
+				elementInstanceList.add(elementInstance);
+			}
+		}
```

Comment:
@wcao20170619 please see Tibor's comment in https://issues.liferay.com/browse/LPS-164513.

Reply:
So I have to use an Element name to identify it and make it available at the backend. Is there any other id available to identify an Element? I guess the front-end also needs some changes for that. @thektan @oliv-yu

Reply:
@wcao20170619 We can possibly use the `externalReferenceCode`. https://github.com/liferay/liferay-portal/blob/master/modules/dxp/apps/search-experiences/search-experiences-service/src/main/resources/com/liferay/search/experiences/internal/model/listener/dependencies/hide_by_exact_term_match.json#L55 

I'll work on the frontend JavaScript changes.